waku/v2/node/peer_manager/peer_manager.nim

```diff
@@ -347,6 +349,17 @@ proc new*(T: type PeerManager,
          maxConnections = maxConnections
     raise newException(Defect, "Max number of connections can't be greater than PeerManager capacity")
 
+  if maxRelayPeers > maxConnections:
+    error "Max number of relay peers can't be greater the max amount of connections",
+         maxConnections = maxConnections,
+         maxRelayPeers = maxRelayPeers
+    raise newException(Defect, "Max number of relay peers can't be greater the max amount of connections")
```

Comment:
This comment applies for further possible enhancements:
What do you think about returning a `Result` rather than raising Exception?

Reply:
yep you are right. not as part of this pr since this just follows how it was done before but definetly something to consider.